src/middlewared/middlewared/plugins/jbof/crud.py

```diff
@@ -304,6 +306,129 @@ async def licensed(self):
                     result += quantity
         return result
 
+    @private
+    @accepts(
+        Int('id', required=True),
+        Str('iom', enum=['IOM1', 'IOM2'], required=True),
+        Dict(
+            'iom_network',
+            Bool('dhcp'),
+            Str('fqdn'),
+            Str('hostname'),
+            List('ipv4_static_addresses', items=[Dict(
+                'ipv4_static_address',
+                IPAddr('address', v6=False),
+                Str('netmask', validators=[Netmask(ipv6=False, prefix_length=False)]),
+                IPAddr('gateway', v6=False))], default=None),
+            List('ipv6_static_addresses', items=[Dict(
+                'ipv6_static_address',
+                IPAddr('address', v4=False),
+                Int('prefixlen', validators=[Range(min_=1, max_=64)]))], default=None),
+            List('nameservers', items=[IPAddr('nameserver')], default=None),
+        ),
+        Int('ethindex', default=1),
+        Bool('force', default=False),
+    )
+    def set_mgmt_ip(self, id_, iom, data, ethindex, force):
+        """Change the mamagement IP for a particular IOM"""
+        # Fetch the existing JBOF config
+        config = self.get_instance__sync(id_)
+        config_mgmt_ips = set([config['mgmt_ip1'], config['mgmt_ip2']])
+
+        # Do we need to switch redfish to the other IOM
+        redfish = self.ensure_redfish_client_cached(config)
+        old_iom_mgmt_ips = set(redfish.iom_mgmt_ips(iom))
+        if redfish.mgmt_ip() in old_iom_mgmt_ips:
+            other_iom = 'IOM2' if iom == 'IOM1' else 'IOM1'
+            for mgmt_ip in redfish.iom_mgmt_ips(other_iom):
+                if mgmt_ip in config_mgmt_ips:
+                    redfish = self.ensure_redfish_client_cached({'mgmt_ip1': mgmt_ip,
+                                                                 'mgmt_username': config['mgmt_username'],
+                                                                 'mgmt_password': config['mgmt_password']})
+                    break
+
+        if not force:
+            if redfish.mgmt_ip() in redfish.iom_mgmt_ips(iom):
+                raise CallError('Can not modify IOM network config thru same IOM')
+
+        # Read the existing config via redfish
+        uri = f'/redfish/v1/Managers/{iom}/EthernetInterfaces/{ethindex}'
+        r = redfish.get(uri)
+        if not r.ok:
+            raise CallError('Unable to read existing network configuration of {iom}/{ethindex}')
+        orig_net_config = r.json()
+
+        newdata = {}
+        olddata = {}
+        if 'dhcp' in data:
```

Comment:
```suggestion
        if dhcp := data.get('dhcp'):
            new data.update({'DHCPv4': {'DHCPenabled': dhcp}})
```

Reply:
I get the idea, but since the `if` has to execute whether or not the value is `True` or `False`, so long as it is present, I think the code has to be:
```
        if dhcp := data.get('dhcp') is not None:
            new data.update({'DHCPv4': {'DHCPenabled': dhcp}})
```
Ditto for the below.

Reply:
You're right 👍